### PR TITLE
Support for sorting/modifying generated statemens

### PR DESCRIPTION
**Pull request by paranoiq**
- allow user to send sorted before/after statements to MigrationService::generateMigrationFile()
- allow user to supply MigrationAnalyzer to sort/modify statements
- fully BC

### CODE REVIEW

File: src/MigrationDummyAnalyzer.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php declare(strict_types = 1);
+
+namespace ShipMonk\Doctrine\Migration;
+
+class MigrationDummyAnalyzer implements MigrationsAnalyzer
```
Comment (janedbal) [inline src/MigrationDummyAnalyzer.php:5]: I'd call it `MigrationDefaultAnalyzer`, similar to `MigrationDefaultVersionProvider` and `MigrationDefaultExecutor`

**Review by janedbal — APPROVED** [review]

File: src/MigrationDummyAnalyzer.php (hunk 1 above)
Comment (paranoiq) [reply src/MigrationDummyAnalyzer.php:5]: i understand, but the difference is that DefaultExecutor and DefaultVP actually do something

Comment (janedbal) [reply src/MigrationDummyAnalyzer.php:5]: The default one does too: "use before stage for all sqls".

Comment (paranoiq) [reply src/MigrationDummyAnalyzer.php]: ok then : ]

**Pull request merged**